config/data/langs.toml

```diff
@@ -22,6 +22,24 @@ each argument is NULL terminated.
 x is a no-op.
 '''
 
+[APL]
+args       = [ '/usr/bin/apl', '-' ]
+experiment = 135
+size       = '14.5 MiB'
+version    = 'Dyalog APL 19.0.48958'
+website    = 'https://www.dyalog.com'
+example    = '''
+⍝ Printing
+⎕←'Hello, World!'
+
+⍝ Looping
+⎕IO←0
+⎕←¨⍳10
+
+⍝ Accessing arguments
+⎕←¨1↓2⎕NQ#'GetCommandLineArgs'
```

Comment:
If you look at the output in the difference field, you'll come to notice that it's actually spreading the output on individual lines by carriage returns. I didn't notice at first, but yet here I am. Now, the sample's always going to be incorrect, but the actual output matches all other languages' visual appearance if I leave it as is.

![image](https://github.com/user-attachments/assets/27d72f4a-a1f9-48d3-bd98-332e81f9c63f) ![image](https://github.com/user-attachments/assets/03409827-b14c-4780-a696-26f110e6c884)

I've written a new, unpushed sample that separates with actual new lines instead of carriage returns. However, while the visual appearance in the difference field is then as should be, regardless of the expectation, the actual output seems to separate printing, looping, and accessing arguments with an additional new line, as if they're paragraphs.

![image](https://github.com/user-attachments/assets/d39268f4-fc4f-48f8-801a-a028e535d6d0)![image](https://github.com/user-attachments/assets/e035ef4d-334f-4192-93a1-af4accf219eb)

I already told you that I know JS about APL, and JS ain't JavaScript, so I figure it might just be a language thing like the index origin. What to do? Me personally, I say as is.

Reply:
I've complained about the carriage returns, but it's probably going to take a while until that changes, if at all. I think for now we should do a bit of replacement on the output to make it work anyway:

```diff
diff --git a/hole/play.go b/hole/play.go
--- hole/play.go
+++ hole/play.go
@@ -447,8 +447,12 @@
 	// Postprocess sed output to turn null bytes into newlines.
 	if lang.ID == "sed" {
 		stdoutBytes = bytes.ReplaceAll(stdoutBytes, []byte("\x00"), []byte("\n"))
 	}
+	// Postprocess apl output to turn carriage return into newlines.
+	if lang.ID == "apl" {
+		stdoutBytes = bytes.ReplaceAll(stdoutBytes, []byte("\r"), []byte("\n"))
+	}
 
 	// Trim trailing whitespace on each line, and then trailing empty lines.
 	// Quine solutions are obviously left untouched.
 	if hole.ID == "quine" {
```

Reply:
@ovs-code

Nice, that seems reasonable.

What's your git commit e-mail address?

Reply:
17097105+ovs-code@users.noreply.github.com